### DISCUSSION TITLE
Can we use the pretrained models on the original pwgan repo

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#204] [Archived]
<USER>>>> georroussos<br/> <DATE> [May 16, 2020, 5:18pm]<br/> <br/> Hi, *slash* <br/> There are many pretrained vocoders on the original repo and some of them<br/> are on LibriTTS. They are very good quality. I wonder if anyone has used<br/> them with Mozilla TTS as is.<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/can-we-use-the-pretrained-models-on-the-original-pwgan-repo]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [May 17, 2020, 11:41am]<br/> <br/> no they use different normalization method for network inputs. But if<br/> you train a new TTS model using mean-var normalization then you can use<br/> their models. Which is possible with dev branch.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [May 17, 2020, 1:59pm]<br/> <br/> Ah, that is such a shame<br/> Some of them are trained for 1M steps and when I performed analysis and<br/> synthesis using the LibriTTS one, it sounded so good! I am now<br/> finetuning the universal WaveRNN you once trained, for 22050Hz instead<br/> and I will share it if anyone is interested<br/> That will hopefully make it easier to apply on more models. On 16kHZ<br/> models it performs **super** nicely, but on 22kHZ there is some noise. I<br/> think I should finetune more. How much more do you think I should do? I<br/> am finetuning on LibriTTS.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [May 17, 2020, 2:10pm]<br/> <br/> It is hard to guess. Just listen the generated audios and see the<br/> quality.<br/> <br/> You can consider to train PWGAN using my branch on LibriTTS. You can<br/> even finetune their released model for faster convergence.<br/> <br/> Another option so to renormalize the spectrograms before PWGAN using<br/> their method at inference. That might also work.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [May 17, 2020, 2:23pm]<br/> <br/> I had the same idea, to finetune their model, in order to make it<br/> compatible with your fork but I think I messed it up. In short, I<br/> couldn't find the correct yaml, because if I cloned the repo using the<br/> latest commit, it gave me the yaml for MelGAN and then, if I checked out<br/> the PWGAN commit, it gives me the ttsv1 and ttsv2 configs, but not the<br/> melgan one in bin/configs, which is needed for feature extraction and I<br/> guess training. Because the melgan config would train MelGAN, wouldn't<br/> it? Also, are these features for both PWGan and MelGAN? I got confused<br/> otherwise yes, I can totally try it!! But I don't know what the correct<br/> workflow and configs are.<br/> Their LibriTTS vocoder is extremely good.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [May 17, 2020, 3:20pm]<br/> <br/> You can take any config from anywhere and run with my branch. Configs<br/> are compatible.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [May 17, 2020, 3:43pm]<br/> <br/> Cool! I will try using the config that came with the model<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [May 18, 2020, 11:08am]<br/> <br/> Which commit do I use to train? **fca88f9** doesn't have the config for<br/> preprocessing and the latest one doesn't have the tts configs. The one<br/> in configs is MelGAN.<br/> <br/> I tried the configs from the original repo but they didn't work.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [May 18, 2020, 12:06pm]<br/> <br/> didn't work means? what was the error?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [May 18, 2020, 12:15pm]<br/> <br/> First I tried to extract the feats using this config<br/> <https://github.com/erogol/ParallelWaveGAN/blob/tts/parallel_wavegan/configs/melgan.v3.long.tts.yaml><br/> <br/> But this is for Melgan training and I wanna train PWGAN<br/> <br/> So I checked out **fca88f9** to get the tts configs. But these configs<br/> are not like the one in the original PWGAN repo,<br/> <https://github.com/kan-bayashi/ParallelWaveGAN/blob/master/egs/libritts/voc1/conf/parallel_wavegan.v1.long.yaml><br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [May 18, 2020, 12:17pm]<br/> <br/> have you tried configs directly from the origina repo?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [May 18, 2020, 12:20pm]<br/> <br/> Yep, the one here<br/> <https://github.com/kan-bayashi/ParallelWaveGAN/blob/master/egs/libritts/voc1/conf/parallel_wavegan.v1.long.yaml><br/> which accompanies the LibriTTS model I want to use for finetuning, but<br/> it seems that the parameters in the yaml file are sprinkled all around<br/> and have different formatting<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [May 18, 2020, 12:21pm]<br/> <br/> which parameter for instance?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [May 18, 2020, 12:28pm]<br/> <br/> I think the config in the original repo doesn't have the datasets<br/> section found in the tts configs, and the audio section in the tts<br/> configs has options that the other one doesn't have. I tried to add the<br/> whole of it, but it didn't work. The product for upsample scales is<br/> computed using 4 values in the original config and 3 in your config. And<br/> the original has a section for STFT losses.<br/> <br/> stft_loss_params: *slash* <br/> fft_sizes: *slash* [1024, 2048, 512 *slash* ] *slash* # List of FFT size for STFT-based loss. *slash* <br/> hop_sizes: *slash* [120, 240, 50 *slash* ] *slash* # List of hop size for STFT-based loss *slash* <br/> win_lengths: *slash* [600, 1200, 240 *slash* ] *slash* # List of window length for STFT-based<br/> loss. *slash* <br/> window: 'hann_window'<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [May 18, 2020, 12:32pm]<br/> <br/> what you can quite easily do is that take the original config and change<br/> the fields as necessary by my fork.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [May 18, 2020, 12:39pm]<br/> <br/> I will try again. What config should I be using for feature extraction?<br/> The melgan one in configs?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [May 18, 2020, 12:41pm]<br/> <br/> for future extraction only thing matters is the audio parameters. just<br/> copy and paste the them from the melgan config to the config you like to<br/> use. You could all figure these by reading the feature extraction code.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [May 19, 2020, 4:50pm]<br/> <br/> I was able to get it to work after all<br/> I took the PWGAN+TTS notebook, here<br/> <https://colab.research.google.com/drive/1cpofjnfKSpFhiREgExENIsum4MrqxyPR><br/> 's fork, since that<br/> is the one I work with, and I changed the synthesize function around. I<br/> was able to load both a trained vocoder using your fork and the vocoders<br/> from the original repo, using a franken'd config.<br/> <br/> Sadly, the quality is not good. I think it is because of what you<br/> mentioned with normalization. The voice comes out as hollow and<br/> distorted, both with the original repo vocoders and the one I tried to<br/> finetune, even though after 30.000 steps (using LibriTTS as finetune)<br/> was producing good speech during eval. Now I will try to finetune the<br/> LJSpeech vocoder you trained for 40.000 steps and see if I get better<br/> results.<br/> <br/> [Archived Post]<br/>
